index.js

```diff
@@ -212,10 +212,9 @@ class ThreadStream extends EventEmitter {
       throw new Error('the worker is ending')
     }
 
-    if (this[kImpl].flushing && this[kImpl].buf.length + data.length >= MAX_STRING) {
+    if (this[kImpl].buf.length + data.length >= MAX_STRING) {
       try {
         writeSync(this)
-        this[kImpl].flushing = true
```

Comment:
This doesn't make sense?

Reply:
It did. There is a flush scheduled in nextTick where this was correctly reset.